src/main/java/io/swagger/inflector/examples/ExampleBuilder.java

```diff
@@ -225,47 +227,51 @@ public static Example fromProperty(Property property, Map<String, Model> definit
-                        prefix = xml.getPrefix();
-                        wrapped = xml.getWrapped();
-                    }
-                }
-                if (model.getExample() != null) {
-                    try {
-                        Example n = Json.mapper().readValue(model.getExample().toString(), Example.class);
-                        output = n;
-                    } catch (IOException e) {
-                        LOGGER.error("unable to convert value", e);
+            if( definitions != null ) {
+                RefProperty ref = (RefProperty) property;
+                Model model = definitions.get(ref.getSimpleRef());
+                if (model != null) {
+                    if (model instanceof ModelImpl) {
+                        ModelImpl i = (ModelImpl) model;
+                        if (i.getXml() != null) {
+                            Xml xml = i.getXml();
+                            name = xml.getName();
+                            attribute = xml.getAttribute();
+                            namespace = xml.getNamespace();
+                            prefix = xml.getPrefix();
+                            wrapped = xml.getWrapped();
+                        }
                     }
-                } else {
-                    ObjectExample values = new ObjectExample();
+                    if (model.getExample() != null) {
+                        try {
+                            Example n = Json.mapper().readValue(model.getExample().toString(), Example.class);
+                            output = n;
+                        } catch (IOException e) {
+                            LOGGER.error("unable to convert value", e);
+                        }
+                    } else {
+                        ObjectExample values = new ObjectExample();
 
-                    Map<String, Property> properties = model.getProperties();
-                    for (String key : properties.keySet()) {
-                        Property innerProp = properties.get(key);
-                        Example p = (Example) fromProperty(innerProp, definitions, processedModels);
-                        if (p != null) {
-                            if (p.getName() == null) {
-                                p.setName(key);
+                        Map<String, Property> properties = model.getProperties();
+                        if (properties != null) {
+                            for (String key : properties.keySet()) {
+                                Property innerProp = properties.get(key);
+                                Example p = (Example) fromProperty(innerProp, definitions, processedModels);
+                                if (p != null) {
+                                    if (p.getName() == null) {
+                                        p.setName(key);
+                                    }
+                                    values.put(key, p);
+                                    processedModels.add(key);
```

Comment:
I can't say why, but I fear we can end up in an infinite loop from this.  I will try to conjure up some test cases to disprove that